runtime/runtime/src/actions.rs

```diff
@@ -30,71 +30,21 @@ use near_primitives::errors::{ActionError, ActionErrorKind, ExternalError, Runti
 use near_vm_errors::{CompilationError, FunctionCallError};
 use near_vm_runner::VMError;
 
-/// Number of epochs it takes to unstake.
-const NUM_UNSTAKING_EPOCHS: u64 = 3;
-
-fn cost_per_block(
-    account_id: &AccountId,
-    account: &Account,
-    runtime_config: &RuntimeConfig,
-) -> Balance {
-    let account_length_cost_per_block = if account_id.len() > 10 {
-        0
-    } else {
-        runtime_config.account_length_baseline_cost_per_block
-            / 3_u128.pow(account_id.len() as u32 - 2)
-    };
-
-    let storage_cost_per_block = u128::from(total_account_storage(account_id, account))
-        * runtime_config.storage_cost_byte_per_block;
-
-    account_length_cost_per_block + storage_cost_per_block
-}
-
-/// Returns Ok if the account has enough balance to pay storage rent for at least required number of blocks.
+/// Returns Ok if the account has enough balance to cover storage stake.
 /// Otherwise returns the amount required.
-/// Validators must have at least enough for `NUM_UNSTAKING_EPOCHS` * epoch_length of blocks,
-/// regular users - `poke_threshold` blocks.
-pub(crate) fn check_rent(
-    account_id: &AccountId,
+pub(crate) fn check_storage_cost(
     account: &Account,
     runtime_config: &RuntimeConfig,
-    epoch_length: BlockHeightDelta,
-) -> Result<(), u128> {
-    let buffer_length = if account.locked > 0 {
-        epoch_length * (NUM_UNSTAKING_EPOCHS + 1)
-    } else {
-        runtime_config.poke_threshold
-    };
-    let buffer_amount =
-        u128::from(buffer_length) * cost_per_block(account_id, account, runtime_config);
-    if account.amount >= buffer_amount {
+) -> Result<(), Balance> {
+    let required_amount =
+        Balance::from(account.storage_usage) * runtime_config.storage_amount_per_byte;
```

Comment:
You need to wrap multiplication to account for malicious state. Every operation in Runtime should be wrapped with the  safe math.

Reply:
Done.